### DISCUSSION TITLE
Memory Leak

### DISCUSSION THREAD

**Discussion by mw-point** [google/or-tools#3084] [CP-SAT questions] [answered]
**What version of OR-Tools and what language are you using?**
ortools: 9.2.9972
Language: Python 3.8.3
 
**Which solver are you using (e.g. CP-SAT, Routing Solver, GLOP, BOP, Gurobi)**
CP-SAT

**What operating system (Linux, Windows, ...) and version?**
Occurred on both Windows (windows 10 Pro, 8 core, 16gb memory) and Linux (ubuntu 20.04 x64 48 core, 196gb memory)

**What did you do?**
Ran my CP-SAT python script

**What did you expect to see**
I expected to see the script run either until OPTIMAL or until it reached the max_time_in_seconds of 600.
I expected to see pretty consistent memory usage of around 1gb - 2gb.

**What did you see instead?**
I saw the script running and consistently increasing memory usage until it was Killed (linux) or stalled (windows).

On Windows, I reverted to ortools v9.0.9048 and was able to run the full 600s without memory going over around 2gb.  I have not tried v9.0 on linux with this current version of my model.

Below is a link to the output of `model.ExportToFile`:
[model.txt](https://github.com/google/or-tools/files/7874833/model.txt)

**Anything else we should know about your project / environment**

**Comment by lperron** [comment]
Can you attach the model?. My intuition is that is is not a memory leak but a large model.

**Reply by mw-point** [reply]
I have attached the output of `model.ExportToFile`.  It's weird that using v9.2 the memory increases steadily, to the point of exhausting all 196gb in about 5 minutes (on the 48-core linux machine), but with v9.0 it stays relatively constant and under about 2gb.

**Comment by lperron** [comment]
Can you also indicate which language you are using.

Thanks

**Reply by mw-point** [reply]
I am using Python

**Comment by lperron** [comment]
Indeed. Thanks for the report.

**Comment by lperron** [comment, marked as answer]
And fixed on master.

Thanks again.
